lightning-invoice/src/utils.rs

```diff
@@ -430,10 +434,18 @@ fn filter_channels(channels: Vec<ChannelDetails>, min_inbound_capacity_msat: Opt
 	// If all channels are private, return the route hint for the highest inbound capacity channel
 	// per counterparty node. If channels with an higher inbound capacity than the
 	// min_inbound_capacity exists, filter out the channels with a lower capacity than that.
+	// Further, if channels which we are connected to the peer for exist, only return such
+	// channels.
```

Comment:
suggestion: "Further, if any channels are online and live, only return such channels."